test/unit/specs/core/network/createNetwork.spec.js

```diff
@@ -101,35 +101,28 @@ describe("createNetwork", () => {
     });
   });
 
-  it("runs onResponseError hook and rejects the returned promise", () => {
-    networkStrategy.and.returnValue(Promise.reject(new Error("myerror")));
-    return network
-      .sendRequest({}, true)
-      .then(() => {
-        throw Error("Expecting sendRequest to fail.");
-      })
-      .catch(error => {
-        expect(lifecycle.onResponseError).toHaveBeenCalledWith(error);
-        expect(error.message).toEqual(
-          "Network request failed.\nCaused by: myerror"
-        );
-      });
+  it("runs onResponseError hook and rejects the returned promise with network error rather than lifecycle error", () => {
+    networkStrategy.and.returnValue(Promise.reject(new Error("networkerror")));
+    lifecycle.onResponseError.and.returnValue(
+      Promise.reject(new Error("lifecycleerror"))
+    );
+    return network.sendRequest({}, true).catch(error => {
+      expect(lifecycle.onResponseError).toHaveBeenCalledWith(error);
+      expect(error.message).toEqual(
+        "Network request failed.\nCaused by: networkerror"
+      );
+    });
   });
 
   it("runs onResponseError hook and rejects the promise when response is invalid json", () => {
     networkStrategy.and.returnValue(Promise.resolve("badbody"));
-    return network
-      .sendRequest({}, true)
-      .then(() => {
-        throw Error("Expecting sendRequest to fail.");
-      })
-      .catch(error => {
-        expect(lifecycle.onResponseError).toHaveBeenCalledWith(error);
-        // The native parse error message is different based on the browser
-        // so we'll just check to parts we control.
-        expect(error.message).toContain("Error parsing server response.\n");
-        expect(error.message).toContain("\nResponse body: badbody");
-      });
+    return network.sendRequest({}, true).catch(error => {
+      expect(lifecycle.onResponseError).toHaveBeenCalledWith(error);
+      // The native parse error message is different based on the browser
+      // so we'll just check to parts we control.
+      expect(error.message).toContain("Error parsing server response.\n");
+      expect(error.message).toContain("\nResponse body: badbody");
+    });
```

Comment:
Technically, there is no test that checks to see whether the promise that is returned by `sendRequest` is chained off the promise that is returned from `onResponseError`. I was on the fence about whether to add a test for that. If anyone feels like I should, let me know and I'll do so.